### PR TITLE
docs: add dedicated account dashboard mention

### PR DESCRIPTION
**Pull request by stevenGravy**
Self-hosted help and support pages link to https://goteleport.com/docs/admin-guides/deploy-a-cluster/deploy-a-cluster/?scope=enterprise#dedicated-account-dashboard but there's no info on downloads. Adding back in.

image

### CODE REVIEW

File: docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx [hunk 1]
```diff
@@ -6,4 +6,10 @@ description: "Guides to running a self-hosted Teleport cluster in production."
 These guides show you how to run a self-hosted Teleport Enterprise cluster in
 production.
 
+## Dedicated account dashboard
```
Comment (ptgott) [inline docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx:9]: This is the only heading in this page, which makes it a little harder for me to read:

image

I would either:

- Remove the heading
- Add another H2, "Guides to self-hosting Teleport", before the `(!toc!)` line.

Comment (stevenGravy) [reply docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx]: Please see update @ptgott

File: docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx [hunk 2]
```diff
@@ -6,10 +6,12 @@ description: "Guides to running a self-hosted Teleport cluster in production."
 These guides show you how to run a self-hosted Teleport Enterprise cluster in
 production.
 
-## Dedicated account dashboard
+# Dedicated account dashboard
```
Comment (ptgott) [inline docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx]:
```suggestion
## Dedicated account dashboard
```
This should be an H2 for SEO purposes (there's already an H1 added based on the title)

File: docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx [hunk 3]
```diff
@@ -6,10 +6,12 @@ description: "Guides to running a self-hosted Teleport cluster in production."
...
 
 Teleport Enterprise subscriptions include a dedicated account dashboard with their preferred
 subdomain of [teleport.sh](https://teleport.sh). The dedicated account dashboard provides
 subscription administrators access to the license file, support links and Teleport Enterprise binary downloads.
 
+# Guides to self-hosting Teleport
```
Comment (ptgott) [inline docs/pages/admin-guides/deploy-a-cluster/deploy-a-cluster.mdx]:
```suggestion
## Guides to self-hosting Teleport 
```

**Review by ptgott — APPROVED** [review]

**Review by scottpgallagher — APPROVED** [review]
Approved.

**Pull request merged**